File: node-app/tictactoe/resultAnalyzer.js
function resultAnalyzer(){
    var obj={};

    obj.board;
	var n="";
	
	obj.result="INPROGRESS";
	var win;
	obj.createResultAnalyzer=function(b) {
		obj.board=b;
	}
	obj.getResult=function(){
		return obj.result;
	}
	obj.getWinner=function(){
		return win;
	}
	obj.checkResult=function(){
		console.log(obj.board.cells[0].getValue);
		
		if(obj.board.cells[0].getValue()!=n&&obj.board.cells[1].getValue()!=n&&obj.board.cells[2].getValue()!=n&&obj.board.cells[3].getValue()!=n&&obj.board.cells[4].getValue()!=n&&obj.board.cells[5].getValue()!=n&&obj.board.cells[6].getValue()!=n&&obj.board.cells[7].getValue()!=n&&obj.board.cells[8].getValue()!=n)
		{
			obj.result="DRAW";
		}
		verticalCheck();
		horizontalCheck();
		diagonalCheck();		
	}
	function diagonalCheck() {
		if(obj.board.cells[0].getValue()=="X"&&obj.board.cells[4].getValue()=="X"&&obj.board.cells[8].getValue()=="X"){
			obj.result="WIN";win=1;
		}
		else if(obj.board.cells[2].getValue()=="X"&&obj.board.cells[4].getValue()=="X"&&obj.board.cells[6].getValue()=="X"){
			obj.result="WIN";win=1;
		}
		
		if(obj.board.cells[0].getValue()=="0"&&obj.board.cells[4].getValue()=="0"&&obj.board.cells[8].getValue()=="0"){
			obj.result="WIN";win=2;
		}
		else if(obj.board.cells[2].getValue()=="0"&&obj.board.cells[4].getValue()=="0"&&obj.board.cells[6].getValue()=="0"){
			obj.result="WIN";win=2;
		}
	}
	function horizontalCheck(){
		if(obj.board.cells[0].getValue()=="X"&&obj.board.cells[1].getValue()=="X"&&obj.board.cells[2].getValue()=="X"){
			obj.result="WIN";win=1;
		}
		else if(obj.board.cells[3].getValue()=="X"&&obj.board.cells[4].getValue()=="X"&&obj.board.cells[5].getValue()=="X"){
			obj.result="WIN";win=1;
		}
		else if(obj.board.cells[6].getValue()=="X"&&obj.board.cells[7].getValue()=="X"&&obj.board.cells[8].getValue()=="X"){
			obj.result="WIN";win=1;
		}
		
		if(obj.board.cells[0].getValue()=="0"&&obj.board.cells[1].getValue()=="0"&&obj.board.cells[2].getValue()=="0"){
			obj.result="WIN";win=2;
		}
		else if(obj.board.cells[3].getValue()=="0"&&obj.board.cells[4].getValue()=="0"&&obj.board.cells[5].getValue()=="0"){
			obj.result="WIN";win=2;
		}
		else if(obj.board.cells[6].getValue()=="0"&&obj.board.cells[7].getValue()=="0"&&obj.board.cells[8].getValue()=="0"){
			obj.result="WIN";win=2;
		}
	}
	function verticalCheck(){
		if(obj.board.cells[0].getValue()=="X"&&obj.board.cells[3].getValue()=="X"&&obj.board.cells[6].getValue()=="X"){
			obj.result="WIN";
			win=1;
		}
		else if(obj.board.cells[1].getValue()=="X"&&obj.board.cells[4].getValue()=="X"&&obj.board.cells[7].getValue()=="X"){
			obj.result="WIN";
			win=1;
		}
		else if(obj.board.cells[2].getValue()=="X"&&obj.board.cells[5].getValue()=="X"&&obj.board.cells[8].getValue()=="X"){
			obj.result="WIN";
			win=1;
		}
		
		if(obj.board.cells[0].getValue()=="0"&&obj.board.cells[3].getValue()=="0"&&obj.board.cells[6].getValue()=="0"){
			obj.result="WIN";
			win=2;
		}
		else if(obj.board.cells[1].getValue()=="0"&&obj.board.cells[4].getValue()=="0"&&obj.board.cells[7].getValue()=="0"){
			obj.result="WIN";
			win=2;
		}
		else if(obj.board.cells[2].getValue()=="0"&&obj.board.cells[5].getValue()=="0"&&obj.board.cells[8].getValue()=="0"){
			obj.result="WIN";
			win=2;
		}
	}

    return obj;
}
module.exports=resultAnalyzer;

File: TicTacToe/app/src/main/java/com/techlabs/paresh/tic_tac_toe/ResultAnalyzer.java
package com.techlabs.paresh.tic_tac_toe;

public class ResultAnalyzer {
	Board board;
	Mark n=Mark.EMPTY;
	
	public State result=State.INPROGRESS;
	int win;
	public ResultAnalyzer(Board b) {
		board=b;
	}
	public State getResult(){
		return result;
	}
	public int getWinner(){
		return win;
	}
	public void checkResult(){
		if(board.cells[0].getValue()!=n&&board.cells[1].getValue()!=n&&board.cells[2].getValue()!=n&&board.cells[3].getValue()!=n&&board.cells[4].getValue()!=n&&board.cells[5].getValue()!=n&&board.cells[6].getValue()!=n&&board.cells[7].getValue()!=n&&board.cells[8].getValue()!=n)
		{
			result=State.DRAW;
		}
		verticalCheck();
		horizontalCheck();
		diagonalCheck();
		 
		
		
	}
	private void diagonalCheck() {
		if(board.cells[0].getValue().equals(Mark.CROSS)&&board.cells[4].getValue().equals(Mark.CROSS)&&board.cells[8].getValue().equals(Mark.CROSS)){
			result=State.WIN;win=1;
		}
		else if(board.cells[2].getValue().equals(Mark.CROSS)&&board.cells[4].getValue().equals(Mark.CROSS)&&board.cells[6].getValue().equals(Mark.CROSS)){
			result=State.WIN;win=1;
		}
		
		if(board.cells[0].getValue().equals(Mark.NOUGHT)&&board.cells[4].getValue().equals(Mark.NOUGHT)&&board.cells[8].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;win=2;
		}
		else if(board.cells[2].getValue().equals(Mark.NOUGHT)&&board.cells[4].getValue().equals(Mark.NOUGHT)&&board.cells[6].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;win=2;
		}
	}
	private void horizontalCheck(){
		if(board.cells[0].getValue().equals(Mark.CROSS)&&board.cells[1].getValue().equals(Mark.CROSS)&&board.cells[2].getValue().equals(Mark.CROSS)){
			result=State.WIN;win=1;
		}
		else if(board.cells[3].getValue().equals(Mark.CROSS)&&board.cells[4].getValue().equals(Mark.CROSS)&&board.cells[5].getValue().equals(Mark.CROSS)){
			result=State.WIN;win=1;
		}
		else if(board.cells[6].getValue().equals(Mark.CROSS)&&board.cells[7].getValue().equals(Mark.CROSS)&&board.cells[8].getValue().equals(Mark.CROSS)){
			result=State.WIN;win=1;
		}
		
		if(board.cells[0].getValue().equals(Mark.NOUGHT)&&board.cells[1].getValue().equals(Mark.NOUGHT)&&board.cells[2].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;win=2;
		}
		else if(board.cells[3].getValue().equals(Mark.NOUGHT)&&board.cells[4].getValue().equals(Mark.NOUGHT)&&board.cells[5].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;win=2;
		}
		else if(board.cells[6].getValue().equals(Mark.NOUGHT)&&board.cells[7].getValue().equals(Mark.NOUGHT)&&board.cells[8].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;win=2;
		}
	}
	private void verticalCheck(){
		if(board.cells[0].getValue().equals(Mark.CROSS)&&board.cells[3].getValue().equals(Mark.CROSS)&&board.cells[6].getValue().equals(Mark.CROSS)){
			result=State.WIN;
			win=1;
		}
		else if(board.cells[1].getValue().equals(Mark.CROSS)&&board.cells[4].getValue().equals(Mark.CROSS)&&board.cells[7].getValue().equals(Mark.CROSS)){
			result=State.WIN;
			win=1;
		}
		else if(board.cells[2].getValue().equals(Mark.CROSS)&&board.cells[5].getValue().equals(Mark.CROSS)&&board.cells[8].getValue().equals(Mark.CROSS)){
			result=State.WIN;
			win=1;
		}
		
		if(board.cells[0].getValue().equals(Mark.NOUGHT)&&board.cells[3].getValue().equals(Mark.NOUGHT)&&board.cells[6].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;
			win=2;
		}
		else if(board.cells[1].getValue().equals(Mark.NOUGHT)&&board.cells[4].getValue().equals(Mark.NOUGHT)&&board.cells[7].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;
			win=2;
		}
		else if(board.cells[2].getValue().equals(Mark.NOUGHT)&&board.cells[5].getValue().equals(Mark.NOUGHT)&&board.cells[8].getValue().equals(Mark.NOUGHT)){
			result=State.WIN;
			win=2;
		}
	}

}
